Settings Page

AdityaKBhadragond14 opened the pull request:
This PR introduces the Settings Page (#67), which can be accessed via the Settings button in the side menu bar.
Features Implemented:

    Network Connectivity Check: Option to verify network connectivity status.
    Token Expiry Check Delay: Configurable delay for verifying if the token has expired.
    Periodic Sync Delay: Configurable delay for periodic data synchronization.

User Actions:

    SAVE Button: Saves the modified settings.
    CANCEL Button: Discards any changes made to the settings.

Pending Implementation:

    Enable/Disable Notifications Setting:
    The toggle exists on the settings page, but its functionality is yet to be implemented.

parthfloyd commented:
Great, thanks @AdityaKBhadragond14 for raising this PR. 
- The layout looks good however the margin needs to be adjusted (preferably marginHorizontal=12dp)
- The save/cancel functionality is not working as expected (It is saving regardless of button press & cancel isn't working).

Please feel free to let me know if there's any question regarding this. 
Screenshot & screen-recording
![image](https://github.com/user-attachments/assets/ca95ccd9-1fbf-40e5-88d6-d858e30a9c4b)
[Screen_recording_20241229_002751.webm](https://github.com/user-attachments/assets/b9a1bac1-0f70-42b9-91c5-2480b230a4b0)

AdityaKBhadragond14 commented:
> Great, thanks @AdityaKBhadragond14 for raising this PR.
> 
>     * The layout looks good however the margin needs to be adjusted (preferably marginHorizontal=12dp)
> 
>     * The save/cancel functionality is not working as expected (It is saving regardless of button press & cancel isn't working).
> 
> 
> Please feel free to let me know if there's any question regarding this. Screenshot & screen-recording ![image](https://private-user-images.githubusercontent.com/40067366/399146590-ca95ccd9-1fbf-40e5-88d6-d858e30a9c4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1MzYyNjgsIm5iZiI6MTczNTUzNTk2OCwicGF0aCI6Ii80MDA2NzM2Ni8zOTkxNDY1OTAtY2E5NWNjZDktMWZiZi00MGU1LTg4ZDYtZDg1OGUzMGE5YzRiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjMwVDA1MTkyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM2ZDBmZjM2OGJlMmUzNzFhNDA1NGNkM2UzODhjYTYyZjg4Y2Q0NTA3NDEzNWI4NzA1YmE2ZTg1NGExN2RlM2MmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.xR2xAUUYwtgfJ4rO5cFtflkBsWjIkb1kACIhsZdWLcw) [Screen_recording_20241229_002751.webm](https://github.com/user-attachments/assets/b9a1bac1-0f70-42b9-91c5-2480b230a4b0)

Hi @parthfloyd ,

I had missed to discard the settings on back press. I have added the fix, please check. Also I have adjusted the horizontal margin in the layout.

icrc-fdeniger commented:
Happy new year @AdityaKBhadragond14 
the build failed on https://github.com/icrc/openmrs-android-fhir/actions/runs/12627643835/job/35253372302?pr=101
Could you have a look ?

parthfloyd commented:
> Happy new year @AdityaKBhadragond14 the build failed on https://github.com/icrc/openmrs-android-fhir/actions/runs/12627643835/job/35253372302?pr=101 Could you have a look ?

Hi @icrc-fdeniger  I've fixed the build issue, it was regarding a minor change in `AndroidManifest.xml` 
@AdityaKBhadragond14 you can merge main to resolve the build issue!

AdityaKBhadragond14 commented:
Hi @icrc-fdeniger I've merged the main branch to resolve the build issue.
